src/test/ui/const-generics/min_const_generics/macro.rs

```diff
@@ -0,0 +1,30 @@
+#![feature(min_const_generics)]
+
+struct Example<const N: usize>;
+
+macro_rules! external_macro {
+  () => {{
+    //~^ ERROR expected type
+    const X: usize = 1337;
+    X
+  }}
+}
+
+fn main() {
+  let _ok = Example::<{
+    #[macro_export]
+    macro_rules! gimme_a_const {
+      ($rusty: ident) => {{
+        let $rusty = 3;
+        $rusty
+      }}
+    };
+    gimme_a_const!(run)
+  }>;
```

Comment:
Can you add a - probably separate - test which uses the exported macro in a different scope and/or crate?

It would also be nice to do the same thing inside of `fn test() -> impl Trait<{ macro def }>`
and `fn test(_: impl Trait<{ macro def }>)`.

Reply:
I ended up mashing them all into one test, I can separate it if that's better?

Reply:
I think it would be good to separate this into a passing and a failing test